alphabet/bag.go

```diff
@@ -107,7 +120,9 @@ func (b *Bag) PutBack(letters []MachineLetter) {
 	for _, ml := range letters {
 		b.tileMap[ml]++
 	}
-	b.Shuffle()
+	if b.fixedOrder {
+		b.Shuffle()
+	}
```

Comment:
we want to shuffle only if we are drawing from a bag in "fixed" order, that is either front to back or back to front on the tiles slice. We can shuffle in the other case too, but it's immaterial - shuffling doesn't really do anything since we draw from the bag randomly.